### PR TITLE
HHH-18309 Oracle LimitHandler does not seem to handle maxrows correctly for native queries

### PR DESCRIPTION
**Pull request by mbladel**
https://hibernate.atlassian.net/browse/HHH-18309

----------------------
By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt)
and can be relicensed under the terms of the [LGPL v2.1 license](https://www.gnu.org/licenses/old-licenses/lgpl-2.1.txt) in the future at the maintainers' discretion.
For more information on licensing, please check [here](https://github.com/hibernate/hibernate-orm/blob/main/CONTRIBUTING.md#legal).

----------------------

### CODE REVIEW

**Comment by quaff** [comment]
All `AbstractLimitHandler`s including `Oracle12LimitHandler` simply ignore `query.setFirstResult(0)`, it violate the decision commented by @beikov , see https://hibernate.atlassian.net/browse/HHH-17475?focusedCommentId=115255

**Comment by mbladel** [comment]
> All AbstractLimitHandlers including Oracle12LimitHandler simply ignore query.setFirstResult(0)

@quaff I'm not sure where you're seeing this, but if you take a look at every `AbstractLimitHandler#processSql` implementation, which is the method invoked to add limit/offset native queries, you will see similar checks throughout all of them.  Even `OffsetFetchLimitHandler` (mentioned in the linked Jira) has exactly the same check I'm introducing here for consistency.

**Comment by quaff** [comment]
> > All AbstractLimitHandlers including Oracle12LimitHandler simply ignore query.setFirstResult(0)
> 
> @quaff I'm not sure where you're seeing this, but if you take a look at every `AbstractLimitHandler#processSql` implementation, which is the method invoked to add limit/offset native queries, you will see similar checks throughout all of them. Even `OffsetFetchLimitHandler` (mentioned in the linked Jira) has exactly the same check I'm introducing here for consistency.

https://github.com/hibernate/hibernate-orm/blob/2214c503836d8fc12495f5626d37ec4e466fcf3c/hibernate-core/src/main/java/org/hibernate/dialect/pagination/AbstractLimitHandler.java#L231-L235

This method should returns true but false if `query.setFirstResult(0)`.

**Comment by mbladel** [comment]
> This method should returns true but false if query.setFirstResult(0)

That would be a change in behavior, and it's surely beyond the scope of this PR that addresses inconsistency with one dialect compared to the rest. Feel free to open a separate Jira and start a discussion there.

**Comment by quaff** [comment]
> > This method should returns true but false if query.setFirstResult(0)
> 
> That would be a change in behavior, and it's surely beyond the scope of this PR that addresses inconsistency with one dialect compared to the rest. Feel free to open a separate Jira and start a discussion there.

FYI, I created https://hibernate.atlassian.net/browse/HHH-18461

File: hibernate-core/src/test/java/org/hibernate/orm/test/query/NativeQueryLimitOffsetTest.java [hunk 1]
```diff
@@ -91,6 +92,24 @@ public void testPartialLimitOffsetOnNativeQuery(SessionFactoryScope scope) {
...
+		scope.inTransaction( session -> {
+			List<Long> l = session.createNativeQuery( "select id from Person where name like :name", Long.class )
+					.setParameter( "name", "J%" )
+					.setFirstResult( 0 )
+					.getResultList();
+			assertEquals( 5, l.size() );
+
+			l = session.createNativeQuery( "select id from Person where name like :name", Long.class )
+					.setParameter( "name", "J%" )
+					.setMaxResults( 0 )
```
Comment (quaff) [inline hibernate-core/src/test/java/org/hibernate/orm/test/query/NativeQueryLimitOffsetTest.java:107]: zero `maxResults` will not trigger sql modification and execution, non-zero `maxResults` should be tested also.

**Comment by quaff** [comment]
> > > This method should returns true but false if query.setFirstResult(0)
> > 
> > 
> > That would be a change in behavior, and it's surely beyond the scope of this PR that addresses inconsistency with one dialect compared to the rest. Feel free to open a separate Jira and start a discussion there.
> 
> FYI, I created https://hibernate.atlassian.net/browse/HHH-18461

PR https://github.com/hibernate/hibernate-orm/pull/8750 fixes both `HHH-18461` and `HHH-18309`.

**Pull request merged**